File: VB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace GridViewSharePartialHelper.Controllers
	Public Class HomeController
		Inherits Controller

		Public Function Index() As ActionResult
			Return View()
		End Function

		Public Function GridViewPartial(ByVal gridType As GridType) As ActionResult
			Dim model As Object = Nothing

			If gridType = GridType.Invoices Then
				model = Invoice.GetData()
			ElseIf gridType = GridType.Products Then
				model = Product.GetData()
			End If

			ViewData("gridType") = gridType

			Return PartialView(model)
		End Function

		Public Function Details(ByVal key As Integer) As ActionResult
			Return View(key)
		End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace GridViewSharePartialHelper.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult GridViewPartial(GridType gridType) {
            object model = null;

            if (gridType == GridType.Invoices)
                model = Invoice.GetData();
            else if (gridType == GridType.Products)
                model = Product.GetData();

            ViewData["gridType"] = gridType;

            return PartialView(model);
        }

        public ActionResult Details(int key) {
            return View(key);
        }
    }
}